cpp/src/io/parquet/experimental/hybrid_scan_impl.hpp

```diff
@@ -0,0 +1,514 @@
+/*
+ * Copyright (c) 2025, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * @file hybrid_scan_impl.hpp
+ * @brief cuDF-IO experimental Parquet reader class implementation header
+ */
+
+#pragma once
+
+#include "hybrid_scan_helpers.hpp"
+#include "io/parquet/parquet_gpu.hpp"
+#include "io/parquet/reader_impl_chunking.hpp"
+
+#include <cudf/io/detail/utils.hpp>
+#include <cudf/io/experimental/hybrid_scan.hpp>
+#include <cudf/io/types.hpp>
+#include <cudf/utilities/memory_resource.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/mr/device/device_memory_resource.hpp>
+
+#include <memory>
+#include <optional>
+#include <vector>
+
+namespace cudf::io::parquet::experimental::detail {
+
+/**
+ * @brief Implementation for Parquet reader
+ */
+class impl {
```

Comment:
`class impl` almost identical to the `reader::impl` class in `parquet/reader_impl.hpp`